safe-global/web-core
# [Create Safe] Handle wallet connection

## What it solves

Resolves #1080 

## How this PR fixes it

- Removes the Cancel and Next button in the first step
- Removes the Back button in the second step
- Navigates the user to the first step if they disconnect their wallet
- Disables the Next button and displays an info box if the connected wallet is on the wrong network
- Always populates the first owner with the connected wallet

### Other changes

- Removed the `useIsConnected` hook again since it isn't being used anymore
- Removed RHF from the third step since it is read-only
- Takes `formState` into account for disabling the Next button

## How to test it

1. Create a new safe
2. Connect a wallet
3. Observe being navigated to step 2
4. No Back button visible
5. Disconnect wallet
6. Observe being navigated to step 1
7. No Cancel or Next button visible
8. Connect wallet that is on a different network
9. Observe being navigated to step 2 with the info box showing
10. Observe the Next button being disabled

## Screenshots

<img width="805" alt="Screenshot 2022-11-07 at 17 01 53" src="https://user-images.githubusercontent.com/5880855/200356773-4aefe914-3365-4655-8c5c-5b937ba4a35d.png">

<img width="788" alt="Screenshot 2022-11-07 at 17 01 41" src="https://user-images.githubusercontent.com/5880855/200356737-aa25fbb7-3e39-47a2-83c4-d235077a093c.png">
